app/controllers/DeleteSchemeChangesController.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.FrontendAppConfig
+import connectors.{PensionSchemeVarianceLockConnector, UpdateSchemeCacheConnector}
+import controllers.actions._
+import forms.DeleteSchemeChangesFormProvider
+import javax.inject.Inject
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.libs.json.{JsError, JsSuccess}
+import play.api.mvc.{Action, AnyContent, Result}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.deleteSchemeChanges
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DeleteSchemeChangesController @Inject()(
+                                               appConfig: FrontendAppConfig,
+                                               override val messagesApi: MessagesApi,
+                                               updateConnector: UpdateSchemeCacheConnector,
+                                               lockConnector: PensionSchemeVarianceLockConnector,
+                                               authenticate: AuthAction,
+                                               getData: DataRetrievalAction,
+                                               requireData: DataRequiredAction,
+                                               formProvider: DeleteSchemeChangesFormProvider
+                                             )(implicit val ec: ExecutionContext) extends FrontendController with I18nSupport with Retrievals {
+
+  private lazy val overviewPage = Redirect(routes.SchemesOverviewController.onPageLoad())
+  private lazy val postCall = routes.DeleteSchemeChangesController.onSubmit _
+  private val form: Form[Boolean] = formProvider()
+
+  def onPageLoad(srn: String): Action[AnyContent] = (authenticate andThen getData).async {
+    implicit request =>
+      getSchemeName(srn) { schemeName =>
+        Future.successful(Ok(deleteSchemeChanges(appConfig, form, schemeName, postCall(srn))))
+      }
+  }
+
+  private def getSchemeName(srn: String)(f: String => Future[Result])
```

Comment:
I'd call this something like "withSchemeName" since you're using the scheme name to do something. Also I'd call the function something like "codeBlock" instead of f